### DISCUSSION TITLE
How to flash yocto built image into Orin Nano devkit hosting Orin NX SoM with external NVMe

### DISCUSSION THREAD

**Discussion by hutcruchi** [OE4T/meta-tegra#1363] [Q&A]
I just built my first yocto image: demo-image-sato.
I have the Orin Nano devkit hosting Orin NX SoM with external NVMe.
When using Nvidia default BSP I used their "initrd flash method" to flah the built image directly to the external NVMe.

Few questions please regarding using alternative tools to the ones provided by Nvidia BSP:

1. Where are the yocto built objects located on my host?
- built image
- device tree blobs
- rootfs
- configuration files 

2. What is the replacement method for flashing the image on my devkit?

3. How can I flash specific partition (for example dtb instead the whole NVMe)

4. How can I modify the partitions layout (set APP partition size or custom parition)

Thanks

**Comment by dwalkes** [comment]
Hi @hutcruchi 
Regarding
> Where are the yocto built objects located on my host?

See https://docs.yoctoproject.org/ref-manual/structure.html and https://docs.yoctoproject.org/ref-manual/structure.html#build-tmp-deploy-images.  The `build/tmp` directory is where most of the build artifacts end up, so `find`s from there usually are helpful for answering these types of questions.

> When using Nvidia default BSP I used their "initrd flash method" to flah the built image directly to the external NVMe
> What is the replacement method for flashing the image on my devkit?

See the wiki page [here](https://github.com/OE4T/meta-tegra/wiki/initrd-flashing-support) which is the [first wiki result in the search for initrd flash](https://github.com/search?q=repo%3AOE4T%2Fmeta-tegra+initrd+flash&type=wikis).  In general you'll find a wealth of information through github searches on the OE4T org, although sometimes you do need to dig.

> How can I flash specific partition (for example dtb instead the whole NVMe)

We don't have comprehensive documentation for this yet, you can find some hints about how people have done this with some searches of discussion posts. Docs updates are welcome if you are so inclined.

> How can I modify the partitions layout (set APP partition size or custom parition)

We don't have a generic description for this either, but specifically for the rootfs size customization you can find some instructions at https://github.com/OE4T/meta-tegra/wiki/initrd-flashing-support#customizing-external-storage-size.
